edx_api_doc_tools/view_utils.py

```diff
@@ -34,14 +33,23 @@ def schema_for_inner(view_class):
         """
         Decorate a view class with the specified schema.
         """
-        decorated_class = method_decorator(
-            name=method_name,
-            decorator=schema(**schema_kwargs),
-        )(view_class)
-        func_docstring = docstring or view_class.__doc__
-        view_func = six.get_unbound_function(getattr(decorated_class, method_name))
-        view_func.__doc__ = func_docstring
-        return decorated_class
+        original_func = getattr(view_class, method_name)
+
+        def new_func(*args, **kwargs):
+            """
+            Defer to the original function.
+
+            This allows us to change the docstring and call `schema` without
+            modifying the original function, which is likely from a DRF library
+            mixin.
+            """
+            return original_func(*args, **kwargs)
+        new_func.__name__ = original_func.__name__
+        new_func.__doc__ = docstring
+        schema(**schema_kwargs)(new_func)
+        setattr(view_class, method_name, new_func)
```

Comment:
Isn't that rather brutal? It makes debugging the view more complicated. Doesn't the docstring problem come from the fact that you were using `six.get_unbound_function`?

Reply:
Yeah, on second thought, I think I'm trying too hard to make the docstring manipulation work.

The root of the issue is that I'm trying to avoid modifying the function at its original definition (which is usually going to be within Django or DRF). The solution I arrived on was declaring a new function, which is a bit contrived.

I pushed a new commit that instead just passes `summary` and `description` to to `@schema` as arguments.